### PR TITLE
[MERGE ON 17 JUNE] EID-1519: Prepare IDP recommendation AB test

### PR DESCRIPTION
**Pull request by JohnWattsGDS**
### **!!!** Merge alphagov/verify-frontend-federation-config#108 first **!!!**

Variant version will present recommended 2 top IDPs more prominently and not
recommended ones a bit less to test user interaction
if the recommendation model would be deployed.

### CODE REVIEW

File: app/controllers/choose_a_certified_company_loa2_variant_controller.rb [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+
+  def index
+    session[:selected_answers]&.delete('interstitial')
+    suggestions = recommendation_engine.get_suggested_idps(current_identity_providers_for_loa, selected_evidence, current_transaction_simple_id)
+    @recommended_idps = IDENTITY_PROVIDER_DISPLAY_DECORATOR.decorate_collection(suggestions[:recommended])
+    @non_recommended_idps = IDENTITY_PROVIDER_DISPLAY_DECORATOR.decorate_collection(suggestions[:unlikely])
+    session[:user_segments] = suggestions[:user_segments]
+    FEDERATION_REPORTER.report_number_of_idps_recommended(current_transaction, request, @recommended_idps.length)
+    # Hack this in here to make this AB test work ... proper config file etc. if the variant wins.
+    top_three = { "post-office-stub" => "post-office-stub", "digidentity" => "digidentity", "experian" => "experian" }
```
[the remark this answers is not in the document]
Comment (JohnWattsGDS) [reply app/controllers/choose_a_certified_company_loa2_variant_controller.rb]: Sorted.

**Comment by JohnWattsGDS** [comment]
Screen Shot 2019-06-05 at 11 54 23
Screen Shot 2019-06-05 at 11 54 59

File: config/locales/en.yml [hunk 2]
```diff
@@ -340,6 +340,12 @@ en:
         title: About %{display_name}
         information_provided_by: Information provided by %{display_name}
         close: close
+      ab_test:
```
Comment (CharlesIC) [inline config/locales/en.yml]: https://github.com/alphagov/verify-frontend/blob/912e9c3ba7af3ee5ce03de09ab62ce95915f44a3/config/locales/en.yml#L343-L348

The [mockups](https://docs.google.com/presentation/d/1eTEt4PaipmWg6hX1hWaW2Ys6tnKAKsiiutZ83xExrWc/edit#slide=id.g33aacf1cdf_0_2) have a different wording - can we please update this to match

File: config/locales/cy.yml [hunk 3]
```diff
@@ -339,6 +339,12 @@ cy:
         title: Am %{display_name}
         information_provided_by: Gwybodaeth a ddarparwyd gan %{display_name}
         close: cau
+      ab_test:
```
Comment (CharlesIC) [inline config/locales/cy.yml]: https://github.com/alphagov/verify-frontend/blob/912e9c3ba7af3ee5ce03de09ab62ce95915f44a3/config/locales/en.yml#L343-L348

The [mockups](https://docs.google.com/presentation/d/1eTEt4PaipmWg6hX1hWaW2Ys6tnKAKsiiutZ83xExrWc/edit#slide=id.g33aacf1cdf_0_2) have a different wording - can we please update this to match

**Comment by CharlesIC** [comment]
I think this looks good now. I'm happy to approve if @jakubmiarka has no objections.

**Comment by CharlesIC** [comment]
We've had three people work and look at this so this should be fine. 
And yes - we'll merge everything in the right order on day one of the test.

**Pull request merged**